lib/vyasa_web/components/contexts/ui_state.ex

```diff
@@ -144,25 +146,151 @@ defmodule VyasaWeb.Context.Components.UiState.Marks do
   end
 end
 
-defmodule VyasaWeb.Context.Components.UiState.Mark do
-  @moduledoc """
-  Definition of ui state corresponding to a single mark.
-  """
-
-  # alias Phoenix.LiveView.Socket
-  alias VyasaWeb.Context.Components.UiState.Mark, as: UiState
-
-  # import Phoenix.Component, only: [assign: 2]
+defmodule VyasaWeb.Context.Components.UiState.Sheaf do
+  alias Vyasa.Sangh.{Sheaf, Mark}
+  alias VyasaWeb.Context.Components.UiState.Marks, as: MarksUiState
+  alias VyasaWeb.Context.Components.UiState.Sheaf, as: SheafUiState
 
+  # QQ: @ks0m1c i haven't thought of whether the is_active should be duplicated from
+  # the actual ui state and kept within the SheafUiState or not.
+  # @rtshkmr to decide later, and remove this flag  if unnecessary
```

Comment:
i duplicated it.

I want to de-couple the active state in the sheaf from this ui active state so that @ init phase, we will have at least one active sheaf (because we use the active value for it). however, we might want more than one "active" (from the pov of ui) sheafs to be displayed.

Reply:
just realised that because state struct is doing tracking, this can be incremental for the current browser sessoin, so we can make it always additive. 

Such patterns will be useful for doing a universal contextual back-button throughout the platform, similar to what redux does in the react world where it's atomic ui changes that can be reloaded / replayed